### PR TITLE
PMT: CAPTCHA for iframe

### PR DESCRIPTION
**Pull request by dfuller114**
# Critical Changes

# Changes
We are now passing a JWT in the Elevate tokenize call in order to bypass CAPTCHA verification in the Elevate iFrame in Gift Entry.

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/classes/GE_PaymentServices.cls
Comment (dfuller114) [inline src/classes/GE_PaymentServices.cls:135]: quick explanation of why `getJwtForToken()` and `getJwtForPurchase()` are being used with some duplicate code. We have to pass an explicit role to get a JWT now since we are generating jwts for more than one role. Since we're using the SDK to generate the JWT for tokenize we can't pass the role as a parameter since that would require Visualforce remoting, which would require the method to be `static` which it isn't and we are using a constructor Builder pattern moving forward for the PS_Request class anyway. That's going to make tokenizing from the SDK a special case. We likely can make the regular `getJwt()` function more generic once we merge with @RadSFDeveloper's code but I have created it this way for now for simplicity sake until we merge with her code.

File: src/classes/GE_PaymentServices.cls
Comment (psadie-sfdo) [inline src/classes/GE_PaymentServices.cls:127]:
```suggestion
            PS_IntegrationServiceConfig.PURCHASE_ROLE);
```

File: src/classes/GE_PaymentServices.cls
Comment (psadie-sfdo) [inline src/classes/GE_PaymentServices.cls:130]:
```suggestion
            jwtPayload.getAsString(),
            config.get(PS_IntegrationServiceConfig.API_KEY)
```

File: src/classes/GE_PaymentServices.cls
Comment (psadie-sfdo) [inline src/classes/GE_PaymentServices.cls:137]:
```suggestion
            PS_IntegrationServiceConfig.TOKEN_ROLE);
```

File: src/classes/GE_PaymentServices.cls
Comment (psadie-sfdo) [inline src/classes/GE_PaymentServices.cls:140]:
```suggestion
            jwtPayload.getAsString(),
            config.get(PS_IntegrationServiceConfig.API_KEY)
```

File: src/classes/GE_PaymentServices.cls
Comment (psadie-sfdo) [inline src/classes/GE_PaymentServices.cls:168]: Should we pluralize `role`? I know it's technically a single string, but multiple roles comma-separated can be contained within this string right?

File: src/pages/GE_TokenizeCard.page
Comment (psadie-sfdo) [inline src/pages/GE_TokenizeCard.page:80]:
```suggestion
            const auth = {
```

**Review by psadie-sfdo — CHANGES_REQUESTED** [review]
Just a few minor comments, for now, pending Radmila's PR merging. Will take another look at that point. Thanks!

File: src/classes/GE_PaymentServices.cls
Comment (RadSFDeveloper) [inline src/classes/GE_PaymentServices.cls:106]: Missing another method:
```
/**
     * @description Returns JSON Web Token to payments services tokenize card iframe
     * @return String
     */
    public String getJwtForToken() {
        return new PS_Request.Builder().getJWT(PS_Request.ElevateEndpoint.TOKENIZE);  
    }
    
```

File: src/classes/PS_IntegrationServiceConfig.cls
Comment (RadSFDeveloper) [inline src/classes/PS_IntegrationServiceConfig.cls:54]: These constants are not required after the latest changes.

File: src/classes/GE_PaymentServices.cls
Comment (dfuller114) [reply src/classes/GE_PaymentServices.cls:106]: @RadSFDeveloper yeah thanks. Was holding off until I made `getJWT()` public in `PS_Request`. Just updated!

File: src/classes/PS_IntegrationServiceConfig.cls
Comment (dfuller114) [reply src/classes/PS_IntegrationServiceConfig.cls]: @RadSFDeveloper updated!

File: src/pages/GE_TokenizeCard.page
Comment (dfuller114) [reply src/pages/GE_TokenizeCard.page]: @psadie-sfdo updated!

File: src/classes/GE_PaymentServices.cls
Comment (dfuller114) [reply src/classes/GE_PaymentServices.cls]: @psadie-sfdo `JWTPayload` isn't being used anymore so we don't have to worry about this one.

File: src/classes/GE_PaymentServices.cls
Comment (dfuller114) [reply src/classes/GE_PaymentServices.cls]: @psadie-sfdo this code has been removed after merging with @RadSFDeveloper's PR.

File: src/classes/GE_PaymentServices.cls
Comment (dfuller114) [reply src/classes/GE_PaymentServices.cls]: this code has been removed after merging with @RadSFDeveloper's PR.

File: src/classes/GE_PaymentServices.cls
Comment (dfuller114) [reply src/classes/GE_PaymentServices.cls]: this code has been removed after merging with @RadSFDeveloper's PR.

File: src/classes/GE_PaymentServices.cls
Comment (dfuller114) [reply src/classes/GE_PaymentServices.cls]: this code has been removed after merging with @RadSFDeveloper's PR.

File: src/classes/GE_PaymentServices.cls
Comment (RadSFDeveloper) [reply src/classes/GE_PaymentServices.cls]: Everything looks good @dfuller114 . Wondering if additional change to [PS_TestPaymentServices](https://github.com/SalesforceFoundation/NPSP/blob/0bd334447467506e0944c62efe5e98c1e9355b04/unpackaged/config/ps_testing/classes/PS_TestPaymentService.cls#L66) is required to handle the CAPTCHA requirements?

File: src/pages/GE_TokenizeCard.page
Comment (ptewson-sfdo) [inline src/pages/GE_TokenizeCard.page:92]: @RadSFDeveloper @dfuller114 - do we want this console log to stay in here? Do we have a plan to keep it from hitting production?

Comment (RadSFDeveloper) [reply src/pages/GE_TokenizeCard.page:92]: Is there any issue in keeping the debug log as is? There is a future work on the RD entry form to provide card holder name to the tokenization, so it is beneficial to keep it for that purpose. However, if there is any issue with the debug statement being included in the release, feel free to remove it in this PR.
CC @force2b

**Review by ptewson-sfdo — APPROVED** [review]
Looks good! Want to remove the console log if possible, but if we make that update, there's no reason to hold up QA!

File: src/pages/GE_TokenizeCard.page
Comment (ptewson-sfdo) [reply src/pages/GE_TokenizeCard.page:92]: We generally don't ship code with log statements, but if it's serving a purpose I'm ok with it remaining on a temporary basis!

Comment (RadSFDeveloper) [reply src/pages/GE_TokenizeCard.page:92]: Since it is a temp debug and it can be added when the future work starts, we can add the console output just for that WI, so @dfuller114 , do you mind removing the debug statement if it is not too late?

Comment (force2b) [reply src/pages/GE_TokenizeCard.page:92]: Agreed. We can just add it back when needed for QA.

**Comment by psadie-sfdo** [comment]
Looks good to me as well! Would also like removal of the console log as Patrick mentioned, but I wouldn't consider it a blocker.

**Review by psadie-sfdo — APPROVED** [review]
Looks good to me as well! Would also like removal of the console log as Patrick mentioned, but I wouldn't consider it a blocker.

File: src/pages/GE_TokenizeCard.page
Comment (dfuller114) [reply src/pages/GE_TokenizeCard.page]: just pushed an update to remove the console log

**Review by RadSFDeveloper — APPROVED** [review]
Looks good, thanks Daniel!

**Pull request merged**
